### PR TITLE
Embeddings

### PR DESCRIPTION
**Pull request by rosequ**
- Supporting word2vec for Anserini
- Updated Java version from 1.7 to 1.8 for Lucene compatibility (pom.xml)

### CODE REVIEW

File: src/main/java/io/anserini/embeddings/search/SearchW2V_Args.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package io.anserini.embeddings.search;
```
Comment (lintool) [inline src/main/java/io/anserini/embeddings/search/SearchW2V_Args.java]: Can you please move this as an inner class, like https://github.com/twitter/GraphJet/blob/master/graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankCassovaryDemo.java

File: src/main/java/io/anserini/embeddings/search/SearchW2V.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.OptionHandlerFilter;
+import org.kohsuke.args4j.ParserProperties;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.Collection;
+
+public class SearchW2V {
+    public static void main(String[] args) {
```
Comment (lintool) [inline src/main/java/io/anserini/embeddings/search/SearchW2V.java]: two space indent, here and everywhere

File: src/main/java/io/anserini/embeddings/train/TrainW2V.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package io.anserini.embeddings.train;
```
Comment (lintool) [inline src/main/java/io/anserini/embeddings/train/TrainW2V.java]: same, inner class.

**Comment by rosequ** [comment]
I have made the necessary changes.

**Pull request merged**